spec/requests/activitypub/inboxes_controller_spec.rb

```diff
@@ -0,0 +1,299 @@
+            'Digest' => digest_header
+          )
+        )
+      end
+      let(:headers) do
+        base_headers.merge(
+          'Digest' => digest_header,
+          'Signature' => signature_header
+        )
+      end
+
+      it 'does not increase the number of accounts' do
+        expect do
+          post "/users/#{local_actor.username}/inbox", params: json.to_json, headers: headers
+        end.to(not_change { Account.count })
+      end
+
+      it 'does not update the remote actors username' do
+        post "/users/#{local_actor.username}/inbox", params: json.to_json, headers: headers
+        expect(response).to have_http_status(202)
+        expect(remote_actor.reload.username).to eq(remote_actor_original_username)
+      end
+    end
+
+    context 'with a follow' do
+      context 'when the remote actor is already following the actor' do
+        before do
+          remote_actor.follow!(local_actor)
+          stub_request(:post, remote_actor_inbox_url)
+        end
+
+        let(:json) do
+          {
+            '@context': 'https://www.w3.org/ns/activitystreams',
+            id: 'https://remote.domain/activities/update/1',
+            type: 'Follow',
+            actor: remote_actor.uri,
+            object: ActivityPub::TagManager.instance.uri_for(local_actor),
+          }.with_indifferent_access
+        end
+        let(:digest_header) { digest_value(json.to_json) }
+        let(:signature_header) do
+          build_signature_string(
+            remote_actor_keypair,
+            'https://remote.domain/users/bob#main-key',
+            "post /users/#{local_actor.username}/inbox",
+            base_headers.merge(
+              'Digest' => digest_header
+            )
+          )
+        end
+        let(:headers) do
+          base_headers.merge(
+            'Digest' => digest_header,
+            'Signature' => signature_header
+          )
+        end
+
+        it 'does not increase the number of accounts' do
+          expect do
+            post "/users/#{local_actor.username}/inbox", params: json.to_json, headers: headers
+          end.to(not_change { Account.count })
+        end
+
+        it 'does not increase the number of follows' do
+          expect do
+            post "/users/#{local_actor.username}/inbox", params: json.to_json, headers: headers
+          end.to(not_change { Follow.count })
+        end
+
+        it 'posts an acceptance to the remote actors inbox' do
+          post "/users/#{local_actor.username}/inbox", params: json.to_json, headers: headers
+          expect(
+            a_request(:post, remote_actor_inbox_url)
+          ).to have_been_made.once
+        end
+
+        it 'does not change the local record of the remote actor' do
+          post "/users/#{local_actor.username}/inbox", params: json.to_json, headers: headers
+          expect(remote_actor.reload.username).to eq(remote_actor_original_username)
+        end
+      end
+    end
```

Comment:
Same, I don't think this needs to be included in addition to the other activity types.

Reply:
Ok. I added Update and Follow in as I wasn't sure if there were any slight differences in processing or not. But if you think it's not needed, I'll remove it along with Update.

Reply:
Done https://github.com/mastodon/mastodon/pull/30361/commits/91a76ad1626a6ebfe67b22f5bff8a9acefb372a9